proxy_agent/src/redirector/windows/bpf_prog.rs

```diff
@@ -124,20 +130,78 @@ pub fn attach_bpf_prog() -> i32 {
                     );
                     return EBPF_FIND_PROGRAM_ERROR;
                 }
-                let fd_id = match bpf_program__fd(connect4_program) {
-                    Ok(fd) => fd,
-                    Err(e) => {
-                        logger::write_error(format!("{}", e));
-                        return EBPF_FIND_PROGRAM_ERROR;
+                let compartment_id = 1;
+                let mut link: ebpf_link_t = ebpf_link_t::empty();
+                let mut link: *mut ebpf_link_t = &mut link as *mut ebpf_link_t;
+                let link: *mut *mut ebpf_link_t = &mut link as *mut *mut ebpf_link_t;
+                match ebpf_prog_attach(
+                    connect4_program,
+                    std::ptr::null(),
+                    &compartment_id as *const i32 as *const c_void,
+                    size_of_val(&compartment_id),
+                    link,
+                ) {
+                    Ok(r) => {
+                        if r != 0 {
+                            logger::write_error(format!(
+                                "Failed to attach authorize_connect4 program with error code: {}.",
+                                r
+                            ));
+                            return EBPF_ATTACH_PROGRAM_ERROR;
+                        }
+                        logger::write_information(
+                            "Success attached authorize_connect4 program.".to_string(),
+                        );
+
+                        match bpf_link_disconnect(*link) {
+                            Ok(_r) => {
+                                logger::write_information("Success disconnected link.".to_string());
+
+                                match bpf_link_destroy(*link) {
+                                    Ok(r) => {
+                                        if r != 0 {
+                                            logger::write_error(format!(
+                                                "Failed to destroy link with error code: {}.",
+                                                r
+                                            ));
+                                            return EBPF_ATTACH_PROGRAM_ERROR;
```

Comment:
may I know what's the link used for?
in case of attach suceed, and link disconnect or destroy failed, do we need to return attach succeed and log error, or just take the whole attach as failed?

Reply:
This link is internal from ebpf, we do not need it but for best practices to destroy it. 
According to Windows ebpf team, the original call is a wrap 3p ebpf call and they do destroy, hence we best to do the same.